nf_core/create.py

```diff
@@ -523,24 +527,26 @@ def git_init_pipeline(self):
         Raises:
             UserWarning: if Git default branch is set to 'dev' or 'TEMPLATE'.
         """
-        # Check that the default branch is not dev
+        default_branch = self.default_branch
         try:
-            default_branch = git.config.GitConfigParser().get_value("init", "defaultBranch")
+            default_branch = default_branch or git.config.GitConfigParser().get_value("init", "defaultBranch")
         except configparser.Error:
-            default_branch = None
             log.debug("Could not read init.defaultBranch")
-        if default_branch == "dev" or default_branch == "TEMPLATE":
+        if default_branch in ["dev", "TEMPLATE"]:
             raise UserWarning(
-                f"Your Git defaultBranch is set to '{default_branch}', which is incompatible with nf-core.\n"
-                "This can be modified with the command [white on grey23] git config --global init.defaultBranch <NAME> [/]\n"
-                "Pipeline git repository is not initialised."
+                f"Your Git defaultBranch '{default_branch}' is incompatible with nf-core.\n"
```

Comment:
```suggestion
                f"Your Git defaultBranch '{default_branch}' is incompatible with nf-core, which relies on the defaultBranch being set to 'master'.\n"
```

Reply:
okay, then let's write that it can't be `dev` or `TEMPLATE`

Reply:
I mean, yes, technically there are things where the branch name "master" is hardcoded in nf-core, but that can change.